### DISCUSSION TITLE
Replacing host in URLs returned in response body

### DISCUSSION THREAD

**Discussion by ztepsic** [dotnet/yarp#1376] [Q&A] [answered]
Hi,

I want to have simple reverse-proxy application that will accept request, append some header value, and forward original path and query to the destination host. Response body returned from destination host should be returned through proxy where all URLs with destination host name are replaced with reverse-proxy host name.

This is current configuration which is doing what I want except replacing destination host name with reverse-proxy name in response body.

```json
"ReverseProxy": {
    "Routes": {
      "route1": {
        "ClusterId": "cluster1",
        "Match": {
          "Path": "{**catch-all}"
        },
        "Transforms": [
          {
            "RequestHeader": "HeaderParam",
            "Append": "SomeValue"
          },
        ]
      }
    },
    "Clusters": {
      "cluster1": {
        "Destinations": {
          "destination1": {
            "Address": "https://example.com/"
          }
        }
      }
    }
  }
```

Does anybody have a suggestion how can I achieve replacing destination hostname with reverse-proxy hostname in response body?

Use case example:
1. Request is sent to reverse-proxy https://localhost::7173/some-path
2. reverse-proxy adds header and sends request to https;//example.com/some-path
3. Response body returned from destination host is:
```json
{
    "img": "https://example.com/image_example.png"
}
```
4. Reverse-proxy returns response content to client which initiated request by replacing destination host in response body
```json
{
    "img": "https://localhost:7173/image_example.png"
}
```

**Comment by Tratcher** [comment, marked as answer]
See https://github.com/microsoft/reverse-proxy/issues/1371
